src/inference/gibbs.jl

```diff
@@ -36,25 +36,38 @@ end
 alg = Gibbs(HMC(0.2, 3, :v1), PG(20, :v2))
 ```
 
+One can also pass the number of iterations for each Gibbs component using the following syntax:
+- `alg = Gibbs((HMC(0.2, 3, :v1), n_hmc), (PG(20, :v2), n_pg))`
+- `alg = Gibbs(HMC(0.2, 3, :v1) => n_hmc, PG(20, :v2) => n_pg)`
+where `n_hmc` and `n_pg` are the number of iterations for HMC and PG in each Gibbs iteration.
+
 Tips:
 - `HMC` and `NUTS` are fast samplers, and can throw off particle-based
 methods like Particle Gibbs. You can increase the effectiveness of particle sampling by including
 more particles in the particle sampler.
 """
-struct Gibbs{space, A<:Tuple} <: InferenceAlgorithm
+struct Gibbs{space, A<:Tuple, B<:Tuple} <: InferenceAlgorithm
     algs::A   # component sampling algorithms
-
-    function Gibbs{space, A}(algs::A) where {space, A<:Tuple}
+    iterations::B
+    function Gibbs{space, A, B}(algs::A, iterations::B) where {space, A<:Tuple, B<:Tuple}
         all(isgibbscomponent, algs) || error("all algorithms have to support Gibbs sampling")
-        return new{space, A}(algs)
+        return new{space, A, B}(algs, iterations)
     end
 end
 
-function Gibbs(algs...)
+function Gibbs(algs::InferenceAlgorithm...)
+    iterations = tuple(fill(1, length(algs))...)
     # obtain space of sampling algorithms
     space = Tuple(union(getspace.(algs)...))
+    return Gibbs{space, typeof(algs), typeof(iterations)}(algs, iterations)
+end
 
-    Gibbs{space, typeof(algs)}(algs)
+function Gibbs(args::T...) where {T <: Union{Tuple{InferenceAlgorithm,Int}, Pair{InferenceAlgorithm,Int}}}
```

Comment:
Same here, this defines (or rather redefines `Gibbs()`. We should rather use something like:

```suggestion
function Gibbs(arg::Pair{InferenceAlgorithm,Int}, args::Vararg{Pair{InferenceAlgorithm,Int},N}) where {N}
```

One problem with the current code is also that it seems it is possible to mix tuples and pairs.